sdk/servicebus/service-bus/test/receiveAndDeleteMode.spec.ts

```diff
@@ -647,164 +551,122 @@ describe("Unsupported features in ReceiveAndDelete mode", function(): void {
-    );
-    await deferMessage();
-    await receiver.close();
-    receiver = receiverClient.createReceiver(ReceiveMode.receiveAndDelete);
-    await receiveDeferredMessage();
-  });
-
-  it("Unpartitioned Subscription: No settlement of the message removes message", async function(): Promise<
-    void
-  > {
-    await beforeEachTest(
-      TestClientType.UnpartitionedTopic,
-      TestClientType.UnpartitionedSubscription,
-      undefined,
-      ReceiveMode.peekLock
-    );
-    await deferMessage();
-    await receiver.close();
-    receiver = receiverClient.createReceiver(ReceiveMode.receiveAndDelete);
-    await receiveDeferredMessage();
-  });
-
-  it("Partitioned Queue with Sessions: No settlement of the message removes message", async function(): Promise<
-    void
-  > {
-    await beforeEachTest(
-      TestClientType.PartitionedQueueWithSessions,
-      TestClientType.PartitionedQueueWithSessions,
-      true,
-      ReceiveMode.peekLock
-    );
-    await deferMessage(true);
-    await receiver.close();
-    receiver = receiverClient.createReceiver(ReceiveMode.receiveAndDelete, {
-      sessionId: TestMessage.sessionId
-    });
-    await receiveDeferredMessage();
-  });
-
-  it("Partitioned Subscription with Sessions: No settlement of the message removes message", async function(): Promise<
-    void
-  > {
-    await beforeEachTest(
-      TestClientType.PartitionedTopicWithSessions,
-      TestClientType.PartitionedSubscriptionWithSessions,
-      true,
-      ReceiveMode.peekLock
-    );
-    await deferMessage(true);
-    await receiver.close();
-    receiver = receiverClient.createReceiver(ReceiveMode.receiveAndDelete, {
-      sessionId: TestMessage.sessionId
-    });
-    await receiveDeferredMessage();
-  });
-
-  it("Unpartitioned Queue with Sessions: No settlement of the message removes message #RunInBrowser", async function(): Promise<
-    void
-  > {
-    await beforeEachTest(
-      TestClientType.UnpartitionedQueueWithSessions,
-      TestClientType.UnpartitionedQueueWithSessions,
-      true,
-      ReceiveMode.peekLock
-    );
-    await deferMessage(true);
-    await receiver.close();
-    receiver = receiverClient.createReceiver(ReceiveMode.receiveAndDelete, {
-      sessionId: TestMessage.sessionId
-    });
-    await receiveDeferredMessage();
-  });
-
-  it("Unpartitioned Subscription with Sessions: No settlement of the message removes message", async function(): Promise<
-    void
-  > {
-    await beforeEachTest(
-      TestClientType.UnpartitionedTopicWithSessions,
-      TestClientType.UnpartitionedSubscriptionWithSessions,
-      true,
-      ReceiveMode.peekLock
-    );
-    await deferMessage(true);
-    await receiver.close();
-    receiver = receiverClient.createReceiver(ReceiveMode.receiveAndDelete, {
-      sessionId: TestMessage.sessionId
-    });
-    await receiveDeferredMessage();
-  });
-});
+// Settlement methods(deferring here) don't exist in the received context in ReceiveAndDelete mode
```

Comment:
I was mistaken, I should probably uncomment these `Receive Deferred messages` tests, and defer the messages by receiving them in peekLock mode.

Reply:
Uncommented and updated.